test/resources/3scale_api.go

```diff
@@ -203,3 +208,119 @@ func (r *ThreeScaleAPIClientImpl) CreateProduct(name string) (string, error) {
 	id := strings.Split(href, "/")
 	return id[len(id)-1], nil
 }
+
+func (r *ThreeScaleAPIClientImpl) SetUserAsAdmin(username string, email string, userID string) error {
+
+	url := fmt.Sprintf("%v/p/admin/account/users/"+userID, r.host)
+	formUrl := fmt.Sprintf("%v/p/admin/account/users/%v/edit", r.host, userID)
+
+	// First try to get the CRSF token by requesting the form
+	csrf, err := requestCRSFToken(r.client, formUrl)
+	if err != nil {
+		return err
+	}
+
+	formValues := url2.Values{
+		"user[username]":              []string{username},
+		"user[email]":                 []string{email},
+		"user[role]":                  []string{"admin"},
+		"user[password]":              []string{},
+		"user[password_confirmation]": []string{},
+		"commit":                      []string{"Update+User"},
+		"authenticity_token":          []string{csrf},
+		"_method":                     []string{"patch"},
+	}
+
+	req, err := http.NewRequest(http.MethodPost, url, strings.NewReader(formValues.Encode()))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", r.token))
+
+	resp, err := r.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.New(fmt.Sprintf("expected 200 but got %v", resp.StatusCode))
+	}
+
+	return nil
+}
+
+func (r *ThreeScaleAPIClientImpl) GetUserId(username string) (string, error) {
+	url := fmt.Sprintf("%v/p/admin/account/users", r.host)
```

Comment:
same as above https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/threescale/three_scale_client.go#L113

Reply:
As discussed, the approach taken here was to follow an existing precedent i.e. the approach taken in 3scale_crudl.go